agent/testagent.go

```diff
@@ -155,35 +147,14 @@ func (a *TestAgent) Start(t *testing.T) (err error) {
 		name = "TestAgent"
 	}
 
-	var cleanupTmpDir = func() {
```

Comment:
I don't see where the data dir is cleaned up now.

Reply:
Oh ya, good catch! I forgot to add the `t.Cleanup`!  I'll add it to `testutil.TempDir`